packages/panna-sdk/src/core/onramp/onramp.ts

```diff
@@ -189,3 +191,67 @@ export function getOnrampProviders(countryCode: string): ProviderInfo[] {
     COUNTRY_PROVIDER_MAP[normalizedCountryCode] || [];
   return providers.map((provider) => PROVIDERS[provider]);
 }
+
+/**
+ * Fetches fiat prices for a specific token or all tokens on a given chain
+ *
+ * @param params - Parameters for fetching token fiat prices
+ * @param params.chainId - The chain ID of the token
+ * @param params.tokenAddress - The address of the token (optional) if not provided, fetches prices for all tokens on the chain
+ * @param params.client - The Panna client instance used for authentication
+ * @returns Promise resolving to the token fiat prices
+ * @throws Error if the token address is invalid or network request fails
+ *
+ * @example
+ * ```ts
+ * // Get fiat prices for a specific token
+ * const prices = await getTokenFiatPrices({
+ *   chainId: 1,
+ *   tokenAddress: '0x0000000000000000000000000000000000000000',
+ *   client: pannaClient
+ * });
+ * // Get fiat prices for all tokens on a chain
+ * const allPrices = await getTokenFiatPrices({
+ *   chainId: 1,
+ *   client: pannaClient
+ * });
+ * // Prices will be an array of objects with token details and fiat values
+ * // Example structure:
+ * * [
+ *   {
+ *     chainId: 1,
+ *     address: '0x0000000000000000000000000000000000000000',
+ *     symbol: 'ETH',
+ *     name: 'Ethereum',
+ *     decimals: 18,
+ *     iconUri: 'https://example.com/eth.png',
+ *     prices: {
+ *       USD: 3000,
+ *       EUR: 2500
+ *     }
+ *   },
+ *   {
+ *     chainId: 1,
+ *     address: '0x0000000000000000000000000000000000000001',
+ *     symbol: 'USDT',
+ *     name: 'Tether',
+ *     decimals: 6,
+ *     iconUri: 'https://example.com/usdt.png',
+ *     prices: {
+ *       USD: 1,
+ *       EUR: 0.85
+ *     }
+ *   }
+ * ]
+ */
+export async function getTokenFiatPrices(
+  params: GetTokenFiatPricesParams
+): Promise<TokenFiatPrice[]> {
+  const { chainId, tokenAddress, client } = params;
+
+  return Bridge.tokens({
+    chainId,
+    tokenAddress,
```

Comment:
1. Does this internally handle the translation of `NATIVE_TOKEN_ADDRESS` to `0x0000000000000000000000000000000000000000`? If not, we must handle it.
2. Should we also take care of undefined `tokenAddress` or is it handled internally by Bridge.tokens?